dispatch/static/manager/src/js/components/fields/IntField.js

```diff
@@ -0,0 +1,23 @@
+import React from 'react'
+
+import { TextInput } from '../inputs'
+
+export default class IntField extends React.Component {
+  onChange(val) {
+    val = val.replace(/[^\d]/, '')
+    this.props.onChange(Math.floor(val))
+  }
+
+  render() {
+    return (
+      <TextInput
+        placeholder={this.props.label}
+        value={this.props.data || ''}
+        fill={true}
+        type="text"
```

Comment:
You should be able to use `type="number"`, which I think removes the need for the regex validation above.

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number